lib/utils/optionsMatches.js

```diff
@@ -8,7 +8,7 @@ const matchesStringOrRegExp = require('./matchesStringOrRegExp');
  *
  * @param {{ [x: string]: any; }} options
  * @param {string} propertyName
- * @param {string} input
+ * @param {unknown} input
```

Comment:
[note] The type guard (`typeof input === 'string'`) allows `input` as `unknown`:

https://github.com/stylelint/stylelint/blob/5ba420993673ac384876fcc7652a7b1ec49b7a12/lib/utils/optionsMatches.js#L19